static/src/javascripts/projects/common/utils/closest.js

```diff
@@ -0,0 +1,18 @@
+define(function () {
+    /**
+     * Adds a wrapper around Element.closest()
+     */
+    var matches = 'matches' in Element.prototype ? 'matches' : 'msMatchesSelector';
+    return 'closest' in Element.prototype ? closestNative : closestPolyfill;
+
+    function closestNative(element, selectors) {
+        return element.closest(selectors);
+    }
+
+    function closestPolyfill(element, selectors) {
+        while (element && !element[matches](selectors)) {
```

Comment:
I think (not sure!) there is a bug here. on the first while iteration, we are running `matches` on the specified node, not the first ancestor. So if the specified node happens to contain descendant nodes that match the selector, the polyfill will return the specified node, even when no closest ancestor satisfies the selector.